src/tools/generators.ts

```diff
@@ -245,3 +259,279 @@ export function installGenerators(generators: string[]): string[] {
 
   return changedGenerators
 }
+
+enum Platforms {
+  Ios = "ios",
+  Android = "android",
+  Expo = "expo",
+}
+
+// prettier-ignore
+const APP_ICON_RULES = {
```

Comment:
These are the icon rulesets. Might look ugly in Github PR view due wrapping and prettier being disabled. 

`icons` are unique outputs: so 1 for ios, 4 for android, and 6 for expo.
`sizes` are variations of each icon.